tm2/pkg/db/pebbledb/pebbledb.go

```diff
@@ -0,0 +1,305 @@
+package pebbledb
+
+import (
+	"bytes"
+	goerrors "errors"
+	"path/filepath"
+	"slices"
+
+	"github.com/cockroachdb/pebble"
+
+	"github.com/gnolang/gno/tm2/pkg/db"
+	"github.com/gnolang/gno/tm2/pkg/db/internal"
+)
+
+func init() {
+	dbCreator := func(name string, dir string) (db.DB, error) {
+		return NewPebbleDB(name, dir)
+	}
+	db.InternalRegisterDBCreator(db.PebbleDBBackend, dbCreator, false)
+}
+
+var _ db.DB = (*PebbleDB)(nil)
+
+type PebbleDB struct {
+	db *pebble.DB
+}
+
+func NewPebbleDB(name string, dir string) (*PebbleDB, error) {
+	return NewPebbleDBWithOpts(name, dir, &pebble.Options{})
+}
+
+func NewPebbleDBWithOpts(name string, dir string, o *pebble.Options) (*PebbleDB, error) {
+	dbPath := filepath.Join(dir, name+".db")
+	db, err := pebble.Open(dbPath, o)
+	if err != nil {
+		return nil, err
+	}
+	database := &PebbleDB{
+		db: db,
+	}
+	return database, nil
+}
+
+// Implements DB.
+func (db *PebbleDB) Get(key []byte) []byte {
+	key = internal.NonNilBytes(key)
+	res, closer, err := db.db.Get(key)
+	if err != nil {
+		if goerrors.Is(err, pebble.ErrNotFound) {
+			return nil
```

Comment:
For other implementations, do we propagate a "not found" error, or just return nil?